### PR TITLE
update chart and refactor default workload type

### PR DESCRIPTION
**Pull request by wonderflow**

### CODE REVIEW

File: vela-templates/web-service.cue [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#Template: {
```
Comment (zzxwill) [inline vela-templates/web-service.cue:1]: Let's change the file name later to `webservice.cue` as I notice you changed the corresponding workload name above:)

Btw, It's great to splitting cue section from WorkloadDefintion/TraitDefinition. But I don't find cue file importing in  manifests.

Comment (wonderflow) [reply vela-templates/web-service.cue:1]: yes, not imported yet, will leave to another PR, hope to find some other guy can help.

File: vela-templates/backend.cue [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#Template: {
```
Comment (ryanzhang-oss) [inline vela-templates/backend.cue:1]: how is this going to be used? Are the cue file supposed to be separate from the defintion.yaml or not? We'd better have a consistent pattern.

File: charts/vela-core/templates/metricstraits.yaml [hunk 3]
```diff
@@ -14,32 +14,4 @@ spec:
...
-      	}
-      }
-      containerized: {
-      	name: string
-      	// +usage=specify app image
-      	// +short=i
-      	image: string
-      	// +usage=specify port for container
-      	// +short=p
-      	port: *6379 | int
```
[the remark this answers is not in the document]
Comment (wonderflow) [reply charts/vela-core/templates/metricstraits.yaml:44]: the content here is wrong, not match

File: charts/vela-core/templates/backend.yaml [hunk 4]
```diff
@@ -18,25 +18,22 @@ spec:
       #Template: {
       	apiVersion: "core.oam.dev/v1alpha2"
       	kind:       "ContainerizedWorkload"
```
[the remark this answers is not in the document]
Comment (wonderflow) [reply charts/vela-core/templates/backend.yaml:20]: Yes, let me change it thanks

File: vela-templates/backend.cue (hunk 2 above)
Comment (wonderflow) [reply vela-templates/backend.cue:1]: Not used yet, prepare to use this link in definition

File: charts/vela-core/templates/backend.yaml [hunk 5]
```diff
@@ -18,25 +18,22 @@ spec:
...
+      	metadata:
+      		name: backend.name
       	spec: {
       		containers: [{
-              image: containerized.image
-              name:  containerized.name
-              ports: [{
-                  containerPort: containerized.port
-                  protocol:      "TCP"
-                  name:          "default"
```
[the remark this answers is not in the document]
Comment (wonderflow) [reply charts/vela-core/templates/backend.yaml:29]: Yes，  it's a backend service

**Comment by zzxwill** [comment]
It seems there is a problem with podSpec protocol validatation.
```
➜  /Users/zhouzhengxi/Programming/golang/src/github.com/wonderflow/kubevela/charts/vela-core git:(template) ✗ helm install vela-core .
Error: failed to install CRD crds/standard.oam.dev_containerizeds.yaml: CustomResourceDefinition.apiextensions.k8s.io "containerizeds.standard.oam.dev" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[podSpec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[podSpec].properties[initContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property]


➜  /Users/zhouzhengxi/Programming/golang/src/github.com/wonderflow/kubevela/charts/vela-core git:(template) ✗ k get CustomResourceDefinition | grep -i containerized
containerizedworkloads.core.oam.dev      2020-08-26T04:23:11Z


➜  /Users/zhouzhengxi/Programming/golang/src/github.com/wonderflow/kubevela/charts/vela-core git:(template) ✗ k delete CustomResourceDefinition containerizedworkloads.core.oam.dev
customresourcedefinition.apiextensions.k8s.io "containerizedworkloads.core.oam.dev" deleted


➜  /Users/zhouzhengxi/Programming/golang/src/github.com/wonderflow/kubevela/charts/vela-core git:(template) ✗ helm install vela-core .
Error: failed to install CRD crds/standard.oam.dev_containerizeds.yaml: CustomResourceDefinition.apiextensions.k8s.io "containerizeds.standard.oam.dev" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[podSpec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[podSpec].properties[initContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property]
```

**Review by zzxwill — APPROVED** [review]
/lgtm

**Review by ryanzhang-oss — APPROVED** [review]

**Comment by yuenandi** [comment]
```
Error: failed to install CRD crds/standard.oam.dev_containerizeds.yaml: CustomResourceDefinition.apiextensions.k8s.io "containerizeds.standard.oam.dev" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[podSpec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[podSpec].properties[initContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property]

```
Has this problem been submitted？I have the same.

**Comment by wonderflow** [comment]
Can you help try `helm install vela-core charts/vela-core/ --disable-openapi-validation`
@yuenandi

**Comment by zzxwill** [comment]
As the PR is merged, filed an issue to track and fix it https://github.com/oam-dev/kubevela/issues/284.

**Pull request merged**